### DISCUSSION TITLE
Support for AWS MSK IAM?

### DISCUSSION THREAD

**Discussion by BruceBailey320** [edenhill/librdkafka#4369] [Q&A]
Hi Folks,

Does librdkafka support AWS MSK IAM?  I've been using this library with great success for several years, but I need to move to AWS MSK.  I've seen discussion going back several years about adding this support, but did this ever happen?

Regards,
Bruce

**Comment by lawrenz1** [comment]
I think you can use MSK with this library using other auth methods like SCRAM or mTLS, but not AWS MSK IAM.

from what i've found:
- JVM based clients can use https://github.com/aws/aws-msk-iam-auth
- https://github.com/twmb/franz-go is a go client that supports AWS MSK IAM

**Reply by BruceBailey320** [reply]
Thanks @lawrenz1 I'll try that and see if I can make headway.  At this moment, Java and Go aren't viable options.  The repos you mention may provide useful clues that I can use, however, so that you for that.

**Comment by ivanbishop** [comment]
Hello Bruce, did you track down support for IAM in librdkafka fork?
